PR: https://github.com/odoo/odoo/pull/

From: 406df78f433c5ca1a8244edd1f41f5bcd94c57c1
From: Olivier Dony
Date: 2018-05-18 09:21:46

Structural Changes: 4
Total Changes: 12

[FIX] hr: make field-level groups consistent

a696f84f1fa5bda33f3db028111a908818396421 adds a few extra fields to the
Private information tab of the employee form. As a result, HR personnel
and employees could reasonably expect these fields to be protected in
the same manner as the other private fields. Make it so.

Cherry-Pick of odoo/enterprise@533db05b5e3b8a28028cb9b61d31eb13296daa55

Fixes #18610

================================= pseudo patch: =================================

--- a/addons/hr/models/hr.py
+++ b/addons/hr/models/hr.py
@@ -150,16 +150,16 @@ class Employee(models.Model):
     permit_no = fields.Char('Work Permit No', groups="hr.group_hr_user")
     visa_no = fields.Char('Visa No', groups="hr.group_hr_user")
     visa_expire = fields.Date('Visa Expire Date', groups="hr.group_hr_user")
-    additional_note = fields.Text(string='Additional Note')
+    additional_note = fields.Text(string='Additional Note', groups="hr.group_hr_user")
     certificate = fields.Selection([
         ('bachelor', 'Bachelor'),
         ('master', 'Master'),
         ('other', 'Other'),
-    ], 'Certificate Level', default='master')
-    study_field = fields.Char("Field of Study", placeholder='Computer Science')
-    study_school = fields.Char("School")
-    emergency_contact = fields.Char("Emergency Contact")
-    emergency_phone = fields.Char("Emergency Phone")
+    ], 'Certificate Level', default='master', groups="hr.group_hr_user")
+    study_field = fields.Char("Field of Study", placeholder='Computer Science', groups="hr.group_hr_user")
+    study_school = fields.Char("School", groups="hr.group_hr_user")
+    emergency_contact = fields.Char("Emergency Contact", groups="hr.group_hr_user")
+    emergency_phone = fields.Char("Emergency Phone", groups="hr.group_hr_user")
     km_home_work = fields.Integer(string="Km home-work", groups="hr.group_hr_user")
     google_drive_link = fields.Char(string="Employee Documents", groups="hr.group_hr_user")
     job_title = fields.Char("Job Title")
